sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -431,9 +430,9 @@ object ScalaJSPluginInternal {
           Def.task {
             mainClass.value map { mainCl =>
               val file = (artifactPath in packageScalaJSLauncher).value
-              val moduleKind = scalaJSModuleKind.value
+              val moduleIdentifier = scalaJSModuleIdentifier.value
               IO.write(file,
-                  launcherContent(mainCl, moduleKind, None),
+                  launcherContent(mainCl, moduleIdentifier),
```

Comment:
That was on purpose. `packageScalaJSLauncher` is not scoped by fast/full, so depending on `scalaJSModuleIdentifier` is wrong, as it will create a file with different content base on the current `scalaJSStage`.